include/core/thermal_expansion_model.h

```diff
@@ -110,4 +111,122 @@ class ConstantThermalExpansion : public ThermalExpansionModel
   const double thermal_expansion;
 };
 
+/**
+ * @brief ThermalExpansionPhaseChange Implements a phase-dependant thermal expansion coefficient
+ */
+class ThermalExpansionPhaseChange : public ThermalExpansionModel
+{
+public:
+  /**
+   * @brief Default constructor
+   */
+  ThermalExpansionPhaseChange(const Parameters::PhaseChange phase_change_params)
+    : p_phase_change_params(phase_change_params)
+    , thermal_expansion_s(phase_change_params.thermal_expansion_s)
+    , thermal_expansion_l(phase_change_params.thermal_expansion_l)
+    , T_solidus(phase_change_params.T_solidus)
+    , T_liquidus(phase_change_params.T_liquidus)
```

Comment:
Why store a copy of all parameters + the phase change params? It's like a weird duplication of parameters. I would just store either what is only needed OR the phase change params overall, but not two versions of the same info.

Reply:
Corrected.